include/dfu/flash_img.h

```diff
@@ -20,6 +20,32 @@ struct flash_img_context {
 	struct stream_flash_ctx stream;
 };
 
+/**
+ * @brief Enum which defines all verification algorithms
+ *
+ * Algorithm list:
+ * - SHA-256
+ */
+enum flash_img_check_alg {
+	FLASH_IMG_CHECK_ALG_SHA256,
+};
+
+/**
+ * @brief Structure for verify flash region integrity
+ *
+ * This is used to pass data and select algorithm to be used to check flash
+ * integrity.
+ *
+ * Note: match vector length is fixed and depends on size from hash algorithm
+ * used to verify flash integrity.
+ */
+struct flash_img_check {
+	enum flash_img_check_alg alg;	/** Match algorithm */
```

Comment:
Maybe it would be better to just provide pointer to callback  here. This way if some dev decides to use custom check function,  there would be no need to modify enum and the `flash_img_check` to support it.

Reply:
I think the `dfu` should abstract it and have a unified implementation for those checks and it'd also allow the user to consume this information. I am afraid of adding custom implementation is fragmentation.